testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jsf/unsupportedwar/FaceletsViewMappingsTestCase.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2020 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.as.test.integration.jsf.unsupportedwar;
+
+import java.net.URL;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.hamcrest.CoreMatchers;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * <p>Tests the JSF deployment failure due to UnsupportedOperationException when
+ * <em>javax.faces.FACELETS_VIEW_MAPPINGS</em> is defined with something that
+ * does not include <em>*.xhtml</em>.</p>
+ *
+ * @author Ranabir Chakraborty <rchakrab@redhat.com>
+ *
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class FaceletsViewMappingsTestCase {
+
+    private static final String DEPLOYMENT = FaceletsViewMappingsTestCase.class.getSimpleName();
+
+    @ArquillianResource
+    private URL url;
+
+    @Deployment
+    public static WebArchive deployment() {
+        final Package DEPLOYMENT_PACKAGE = HelloBean.class.getPackage();
+        WebArchive war = ShrinkWrap.create(WebArchive.class, DEPLOYMENT + ".war");
+        war.addClasses(HelloBean.class, ConfigurationBean.class);
+        war.addAsWebInfResource(DEPLOYMENT_PACKAGE, "web.xml", "web.xml");
+        war.addAsWebResource(DEPLOYMENT_PACKAGE, "hello.jsf", "hello.jsf");
+        return war;
+    }
+
+    @SuppressWarnings("deprecation")
+    @Test
+    public void testHelloWorld() throws Exception {
+        try (CloseableHttpClient httpclient = HttpClients.createDefault()) {
+            String Url = url.toExternalForm();
+            HttpGet httpget = new HttpGet(Url);
+            httpclient.execute(httpget);
+
+            HttpResponse response = httpclient.execute(httpget);
+            Assert.assertEquals("Operation failed", HttpStatus.SC_OK, response.getStatusLine().getStatusCode());
```

Comment:
Suggestion, instead of "Operation failed", could you add something more informative? that would helps if you are reviewing the logs. "Operation failed" sounds like there is a management operation involved. In addition, you could add the result of the `response.getStatusLine().getStatusCode()` so you can easily see in the logs what was returned by the previous HTTP request.